services/src/main/java/beans/PlaylistBean.java

```diff
@@ -1,4 +1,78 @@
 package beans;
 
+import core.Playlist;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.TypedQuery;
+import javax.transaction.Transactional;
+import java.util.List;
+
+@ApplicationScoped
 public class PlaylistBean {
+
+    @PersistenceContext(unitName = "catalog-jpa")
+    private EntityManager entityManager;
+
+    public List<Playlist> getPlaylists() {
+        TypedQuery<Playlist> playlists = entityManager.createNamedQuery("Playlist.getAll", Playlist.class);
+        return playlists.getResultList();
+    }
+
+    public Playlist getPlaylistById(int idPlaylist) {
+        return entityManager.find(Playlist.class, idPlaylist);
+    }
+
+    @Transactional
+    public Playlist addPlaylist(Playlist playlist) {
+        if(playlist == null)
+            return null;
+
+        entityManager.persist(playlist);
+        return playlist;
+    }
+
+    @Transactional
+    public Playlist updatePlaylist(Playlist playlist) {
+        if(playlist == null)
+            return null;
+
+        Playlist existingPlaylist = entityManager.find(Playlist.class, playlist.getId());
+
+        if(playlist.getPlaylistName() != null)
+            existingPlaylist.setPlaylistName(playlist.getPlaylistName());
+
+        if(playlist.getSongs().size() != 0)
```

Comment:
Use `isEmpty()` to check whether a collection is empty.
I.e. `if(!playlist.getSongs().isEmpty())`

Reply:
And do the same for the other 2 `if` checks here

Reply:
Changes implemented in c71f00b.